### PR TITLE
[IMP] auth_saml nicer handling of logout and autoredirect

### PR DESCRIPTION
**Pull request by gurneyalex**
When autoredirect is enabled, but some technical accounts have a normal
password authentication setup, it can be difficult to access them as
logging out of a session will trigger the autoredirect feature.

We change this by overriding the logout redirection so that it sets the
disable_autoredirect parameter -> this way the user will get redirected
to the normal signin screen.

### CODE REVIEW

File: auth_saml/controllers/main.py [hunk 1]
```diff
@@ -270,3 +271,20 @@ def saml_metadata(self, req, **kw):
                 ),
                 [("Content-Type", "text/xml")],
             )
+
+
+class Session(WebSession):
+    @http.route("/web/session/logout", type="http", auth="none")
+    def logout(self, redirect="/web/login"):
```
Comment (gurneyalex) [inline auth_saml/controllers/main.py:278]: I'm wondering if I should not just say 
```suggestion
    def logout(self, redirect="/web/login?disable_autoredirect="):
```

and simply  `return super().logout(redirect)`

Comment (yankinmax) [reply auth_saml/controllers/main.py:278]: hello @gurneyalex , reading the doc https://github.com/OCA/server-auth/blob/15.0/auth_saml/README.rst#configuration I think we could try to test it and have an answer

**Review by yankinmax — APPROVED** [review]

**Review by ivantodorovich — APPROVED** [review]

File: auth_saml/controllers/main.py (hunk 1 above)
Comment (yankinmax) [reply auth_saml/controllers/main.py:278]: I've tried this variant, seems we have the same expected result

**Pull request merged**